paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/PurgeFilesProcedure.java

```diff
@@ -45,30 +47,40 @@ public class PurgeFilesProcedure extends ProcedureBase {
 
     public static final String IDENTIFIER = "purge_files";
 
-    @ProcedureHint(argument = {@ArgumentHint(name = "table", type = @DataTypeHint("STRING"))})
-    public String[] call(ProcedureContext procedureContext, String tableId)
+    @ProcedureHint(
+            argument = {
+                @ArgumentHint(name = "table", type = @DataTypeHint("STRING")),
+                @ArgumentHint(name = "dry_run", type = @DataTypeHint("BOOLEAN"), isOptional = true)
+            })
+    public @DataTypeHint("ROW<dir_path STRING>") Row[] call(
+            ProcedureContext procedureContext, String tableId, Boolean dryRun)
             throws Catalog.TableNotExistException {
         Table table = catalog.getTable(Identifier.fromString(tableId));
         FileStoreTable fileStoreTable = (FileStoreTable) table;
         FileIO fileIO = fileStoreTable.fileIO();
         Path tablePath = fileStoreTable.snapshotManager().tablePath();
+        ArrayList<String> deleteDir = new ArrayList<>();
```

Comment:
Minor optizime: consider here may contains much dirs, change to constructor with size maybe better here.